Add vicare burner attributes

stefanwi87 opened the pull request:
ViCare API delivers information about statistics like burner starts (times the burner has been startet since commissioning) and burner hours (hours the burner has been in operation since commissioning). Please add these two more information in the vicare-integration as suggested. PyViCare-changes are already suggested (follow link https://github.com/stefanwi87/PyViCare/tree/patch-1)

Upstream PR: https://github.com/somm15/PyViCare/pull/25

<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Breaking change
<!--
  If your PR contains a breaking change for existing users, it is important
  to tell them what breaks, how to make it work again and why we did this.
  This piece of text is published with the release notes, so it helps if you
  write it towards our users, not us.
  Note: Remove this section if this PR is NOT a breaking change.
-->


## Proposed change
<!-- 
ViCare API delivers information about statistics like burner starts (times the burner has been startet since commissioning) and burner hours (hours the burner has been in operation since commissioning). Please add these two more information in the vicare-integration as suggested. PyViCare-changes are already suggested (follow link https://github.com/stefanwi87/PyViCare/tree/patch-1)
-->


## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box! 
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [x] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example entry for `configuration.yaml`:
<!--
sensor:
  - platform: template
    # ViCare
    sensors:
      burner_starts:
        friendly_name: "Brennerstarts ViCare"
        unit_of_measurement: 'times'
        value_template: "{{ state_attr('climate.vicare_heating', 'burner_starts')}}"
      burner_hours:
        friendly_name: "Brennerstunden ViCare"
        unit_of_measurement: 'times'
        value_template: "{{ state_attr('climate.vicare_heating', 'burner_hours')}}"
-->

```yaml
# Example configuration.yaml

```

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [ ] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

stefanwi87 committed:
Update climate.py

ViCare API delivers information about statistics like burner starts (times the burner has been startet since commissioning) and burner hours (hours the burner has been in operation since commissioning). Please add these two more information in the vicare-integration as suggested. PyViCare-changes are already suggested (follow link https://github.com/stefanwi87/PyViCare/tree/patch-1)

homeassistant commented:
Hi @stefanwi87, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#32304).

Once you do that we will be able to review and accept this pull request.

Thanks!

oischinger reviewed (changes requested):
The PyVicare lib needs to merge your PR and release a new version. Then we have to refer to this version in the Manifest.

Once this is done I'm fine with this PR.

springstan commented:
@stefanwi87 any updates on this? :) FYI your upstream PR has been merged.

oischinger commented on homeassistant/components/vicare/climate.py:
```diff
@@ -162,6 +162,9 @@ def update(self):
+                self._attributes["burner_starts"] = self._api.getBurnerStarts()
 
             elif self._heating_type == HeatingType.heatpump:
                 self._current_action = self._api.getCompressorActive()
```
If you don't mind you could also add
self._attributes["heatpump_supply_temperature"] = self._api.getSupplyTemperaturePrimaryCircuit()
self._attributes["heatpump_return_temperature"] = self._api.getReturnTemperaturePrimaryCircuit()
as requested here: https://community.home-assistant.io/t/viessmann-component/77873/140
Otherwise I'll create a separate PR.

stefanwi87 replied:
Hi @oischinger,

sure, no problem. I've added the lines in the heatpump-else.

Regards,
Stefan

michelde replied:
@oischinger  @stefanwi87 :  In the current master of PyViCare/Heatpump [Lines 20-30](https://github.com/somm15/PyViCare/blob/master/PyViCare/PyViCareHeatPump.py) we also have two new properties for heatpump statistics: compressor starts and hours, which are similar to burner starts and hours. So, could we add them in this PR as well under the heatpump section? This would give HA users the same functionality/information regardless if they use a burner or heatpump.

oischinger reviewed (changes requested):
Blocking until PyPi release of PyVicare is available

stefanwi87 committed:
Update climate.py

stefanwi87 committed:
Update climate.py

Sorry, just forgot to add the following, gas-specific data: 
- Current burner power in kW
- Todays gas consumption for heating
- Todays gas consumption for hot water

oischinger commented:
I asked for a newe PyVicare release: https://github.com/somm15/PyViCare/pull/25

oischinger commented:
PyViCare 0.1.9 release is now out.
Just add  "requirements": ["PyViCare==0.1.9"] to manifest.json and run python3 -m script.gen_requirements_all, then we should be all set

oischinger reviewed (approved):
Unblocking this PR since PyVicare was updated to 1.1.10 on dev branch

oischinger commented:
Rebase is needed.

oischinger commented:
There has been to progress on this PR and we the attributes that have been added here are integrated as sensors in this PR:
https://github.com/home-assistant/core/pull/34385

I suggest to close this one

frenck commented:
As suggested, lets close this one in favor of #34385
